src/controllers/controller.scatter.js

```diff
@@ -1,7 +1,126 @@
+      start = 0;
+      count = points.length;
+    }
+
+    if (this.options.showLine) {
+
+      const {dataset: line, _dataset} = meta;
+
+      // Update Line
+      line._chart = this.chart;
+      line._datasetIndex = this.index;
+      line._decimated = !!_dataset._decimated;
+      line.points = points;
+
+      const options = this.resolveDatasetElementOptions(mode);
+      options.segment = this.options.segment;
+      this.updateElement(line, undefined, {
+        animated: !animationsDisabled,
+        options
+      }, mode);
+    } else {
+      this.options.borderWidth = 0;
+    }
+
+    // Update Points
+    this.updateElements(points, start, count, mode);
+  }
+
+  updateElements(points, start, count, mode) {
+    const reset = mode === 'reset';
+    const {iScale, vScale, _stacked, _dataset} = this._cachedMeta;
+    const firstOpts = this.resolveDataElementOptions(start, mode);
+    const sharedOptions = this.getSharedOptions(firstOpts);
+    const includeOptions = this.includeOptions(mode, sharedOptions);
+    const iAxis = iScale.axis;
+    const vAxis = vScale.axis;
+    const {spanGaps, segment} = this.options;
+    const maxGapLength = isNumber(spanGaps) ? spanGaps : Number.POSITIVE_INFINITY;
+    const directUpdate = this.chart._animationsDisabled || reset || mode === 'none';
+    let prevParsed = start > 0 && this.getParsed(start - 1);
+
+    for (let i = start; i < start + count; ++i) {
+      const point = points[i];
+      const parsed = this.getParsed(i);
+      const properties = directUpdate ? point : {};
+      const nullData = isNullOrUndef(parsed[vAxis]);
+      const iPixel = properties[iAxis] = iScale.getPixelForValue(parsed[iAxis], i);
+      const vPixel = properties[vAxis] = reset || nullData ? vScale.getBasePixel() : vScale.getPixelForValue(_stacked ? this.applyStack(vScale, parsed, _stacked) : parsed[vAxis], i);
+
+      properties.skip = isNaN(iPixel) || isNaN(vPixel) || nullData;
+      properties.stop = i > 0 && (Math.abs(parsed[iAxis] - prevParsed[iAxis])) > maxGapLength;
+      if (segment) {
+        properties.parsed = parsed;
+        properties.raw = _dataset.data[i];
+      }
+
+      if (includeOptions) {
+        properties.options = sharedOptions || this.resolveDataElementOptions(i, point.active ? 'active' : mode);
+      }
+
+      if (!directUpdate) {
+        this.updateElement(point, i, properties, mode);
+      }
+
+      prevParsed = parsed;
+    }
+
+    this.updateSharedOptions(sharedOptions, mode, firstOpts);
+  }
 
+  /**
+	 * @protected
+	 */
+  getMaxOverflow() {
+    const meta = this._cachedMeta;
+    const data = meta.data || [];
+
+    if (!this.options.showLine) {
+      let max = 0;
+      for (let i = data.length - 1; i >= 0; --i) {
+        max = Math.max(max, data[i].size(this.resolveDataElementOptions(i)) / 2);
+      }
+      return max > 0 && max;
+    }
+
+    const dataset = meta.dataset;
+    const border = dataset.options && dataset.options.borderWidth || 0;
```

Comment:
When setting `showLine` to true in the root op the options it seems like dataset is null here might have somehting to do with my above comment

Reply:
@LeeLenaleee, I tuned my solution and fixed this bug. Thank you for noticing it.

Also, I have added tests for showLine parameter in the Scatter controller.

I updated the [sandbox](https://codesandbox.io/s/summer-dew-d7eodw?file=/src/components/ScatterChart.ts) from the PR description also.